File: src/main/scala/s/chain_of_resposibility/like/functional/MessagePrinter.scala
package s.chain_of_resposibility.like.functional

import s.chain_of_resposibility.Trouble

object MessagePrinter {
  def done(name: String, trouble: Trouble): Unit = {
    System.out.println(trouble + " is resolved by [" + name + "].")
  }

  def fail(trouble: Trouble): Unit = {
    System.out.println(trouble.toString + " cannot be resolved.")
  }
}

File: src/main/java/j/chain_of_resposibility/like/functional/MessagePrinter.java
package j.chain_of_resposibility.like.functional;

import j.chain_of_resposibility.Trouble;

public final class MessagePrinter {
  static void done(String name, Trouble trouble) {
    System.out.println(trouble + " is resolved by [" + name + "].");
  }

  static void fail(Trouble trouble) {
    System.out.println(trouble + " cannot be resolved.");
  }
}
